chains/substrate/listener/listener.go

```diff
@@ -59,6 +63,7 @@ func (l *SubstrateListener) ListenToEvents(ctx context.Context, startBlock *big.
 			case <-ctx.Done():
 				return
 			default:
+				ctxWithSpan, span := otel.Tracer("relayer-sygma").Start(ctx, "relayer.sygma.SubstrateListener.ListenToEvents")
```

Comment:
Here it is used in the right way, we should not override ctx here, otherwise, all the span here would be jammed in one big trace while those are separate events (ever listen to incremented block)